### DISCUSSION TITLE
애니메이션의 프레임을 얻고싶습니다.

### DISCUSSION THREAD

**Discussion by ted002** [naverz/zepeto-studio-kor#433] [5. Q&A]
현재 애니메이션의 특정 프레임에서 이벤트를 발생시켜서 몇가지 효과를 일으키고 싶습니다.

하지만 현재 진행해 본 결과 코드내에서 현재 재생되고 있는 애니메이션이 몇번째 프레임인지 알수 있는 방법은 없더라구요.

그래서 애니메이션 내부에 있는 이벤트 기능을 활성화 하려고 하니, 이건 또 C#으로 작성된 코드가 컴포넌트로 필요해서 당연히 안되구요.

방법이 없을까요?

**Comment by Zepeto-TechEvangelist** [comment]
간접적으로 획득하는 방법 안내 드립니다.

var curTime = (ZepetoCharacter) character.ZepetoAnimator.GetCurrentAnimatorStateInfo(0).normalizedTime;
// (현재 재생중인 클립의 normalizeTime(0-1) )

var curClip = (해당 AnimationClip);

curTime * curClip.length * curClip.frameRate;
// (현재 재생중인 클립의 Frame)

**Reply by ted002** [reply]
감사합니다! 잘 써보겟습니다!
